### ISSUE TITLE
Margin from template parts is not visible on select in the Site Editor

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#37106]
When the contents of a template part include margin, that margin is visible when you edit the template. Notice the whitespace shown to indicate the bottom margin here in Twenty Twenty-Two:

<img width="1211" alt="Screen Shot 2021-12-03 at 11 00 44 AM" src="https://user-images.githubusercontent.com/1202812/144633978-200b7c88-6318-4155-b256-16a7ba69c4e0.png">

When you're using the Site Editor though, it's unclear that the margin above is actually part of the template part. When I select the template part, the blue outline excludes the margin: 

<img width="1213" alt="Screen Shot 2021-12-03 at 11 04 18 AM" src="https://user-images.githubusercontent.com/1202812/144633961-aa3e19cc-1818-4241-ad68-1cad1ff7809f.png">

This is a discoverability problem for users. If they'd like to edit that margin, it's hard for them to tell where it's coming from when using the Site Editor. 

cc @jameskoster, I'm curious if you have thoughts here.

**Comment by jameskoster** [comment]
Yeah this feels like a bug to me too. That margin should be visible when editing the template. 

The rise of user-defined block paddings and margins also raises the value of something like https://github.com/WordPress/gutenberg/issues/36437 as an aid to the discoverability issues you highlighted.

As a side-note, we touched on a similar problem earlier today in https://github.com/WordPress/gutenberg/issues/36827 (starting around [here](https://github.com/WordPress/gutenberg/issues/36827#issuecomment-984448714)). You may be interested in the conversation there.

**Comment by kjellr** [comment]
Cool, thanks. We're having some parallel discussion here about whether we should be using Margin for this specific Twenty Twenty-Two use case anyway: https://github.com/WordPress/gutenberg/pull/37105
